Index: Server.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>from socket import *\r\n\r\nserverPort = 12000\r\nserverSocket = socket(AF_INET, SOCK_DGRAM)\r\nserverSocket.bind(('', serverPort))\r\nprint('server is up and ready')\r\nwhile True:\r\n    message = serverSocket.recv(2048).decode()\r\n    print('message {} received'.format(message))\r\n    modifiedMessage = message.upper()\r\n    print('sending server response...')\r\n    serverSocket.sendto(modifiedMessage.encode())\r\n    print('response sent.')\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Server.py b/Server.py
--- a/Server.py	
+++ b/Server.py	
@@ -5,9 +5,9 @@
 serverSocket.bind(('', serverPort))
 print('server is up and ready')
 while True:
-    message = serverSocket.recv(2048).decode()
-    print('message {} received'.format(message))
+    message, clientAddr = serverSocket.recvfrom(2048)
+    print('message "{}" received from {}'.format(message.decode(), clientAddr))
     modifiedMessage = message.upper()
-    print('sending server response...')
-    serverSocket.sendto(modifiedMessage.encode())
+    print('sending server response ({}) to {}'.format(modifiedMessage.decode(), clientAddr))
+    serverSocket.sendto(modifiedMessage, clientAddr)
     print('response sent.')
